compiler/ballerina-lang/spotbugs-exclude.xml

```diff
@@ -354,6 +354,11 @@
         <Method name="extractBala"/>
         <Bug pattern="RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE"/>
     </Match>
+    <Match>
+        <Class name="io.ballerina.projects.internal.model.BalToolDescriptor"/>
+        <Method name="getToolJarsMatchingPattern"/>
+        <Bug pattern="RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE"/>
```

Comment:
We must use `Optional.of` for the path instead of adding a Spotbug exclusion

Reply:
As per the offline discussion, this is due to the usage of try-with-resources. Have to keep the spotbugs exclusion to avoid this.